libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdown.tsx

```diff
@@ -0,0 +1,93 @@
+// (C) 2020 GoodData Corporation
+import React, { useState } from "react";
+import Overlay from "@gooddata/goodstrap/lib/core/Overlay";
+import { IRankingFilter } from "@gooddata/sdk-model";
+import { IntlWrapper } from "@gooddata/sdk-ui";
+import { RankingFilterDropdownBody } from "./RankingFilterDropdownBody";
+import { injectIntl, WrappedComponentProps } from "react-intl";
+import { IMeasureDropdownItem, IAttributeDropdownItem } from "./types";
+import { getFirstLocalIdentifierObj } from "./utils";
+
+const alignPoints = ["bl tl", "tl bl", "br tr", "tr br"];
+const DROPDOWN_ALIGNMENTS = alignPoints.map((align) => ({ align, offset: { x: 1, y: 0 } }));
+
+const prepareRankingFilterState = (
+    filter: IRankingFilter,
+    measureItems: IMeasureDropdownItem[],
+    attributeItems: IAttributeDropdownItem[],
+): IRankingFilter => {
+    const rankingFilter = filter.rankingFilter;
+
+    const firstMeasure = getFirstLocalIdentifierObj(rankingFilter.measures);
+    const firstAttribute = getFirstLocalIdentifierObj(rankingFilter.attributes);
+
+    return {
+        rankingFilter: {
+            ...rankingFilter,
+            measures: firstMeasure ? [firstMeasure] : [measureItems[0]],
```

Comment:
You need to throw error when `firstMeasure` is not found otherwise this will fail on lower level imho.